akka-cluster/src/main/scala/akka/cluster/Member.scala

```diff
@@ -25,7 +25,7 @@ class Member private[cluster] (
   val roles:                     Set[String]) extends Serializable {
 
   lazy val dataCenter: DataCenter = roles.find(_.startsWith(ClusterSettings.DcRolePrefix))
-    .getOrElse(throw new IllegalStateException("DataCenter undefined, should not be possible"))
+    .getOrElse(s"${ClusterSettings.DcRolePrefix}${ClusterSettings.DefaultDataCenter}") // node which is unaware of DC- settings
```

Comment:
How can this happen? The serializer will add the default data center role: https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala#L464-L478